internal/api/impl/v1/rolebinding/service.go

```diff
@@ -0,0 +1,116 @@
+	"github.com/perses/perses/internal/api/shared/schemas"
+	"github.com/perses/perses/pkg/model/api"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/sirupsen/logrus"
+)
+
+type service struct {
+	rolebinding.Service
+	dao rolebinding.DAO
+	sch schemas.Schemas
+}
+
+func NewService(dao rolebinding.DAO, sch schemas.Schemas) rolebinding.Service {
+	return &service{
+		dao: dao,
+		sch: sch,
+	}
+}
+
+func (s *service) Create(entity api.Entity) (interface{}, error) {
+	if object, ok := entity.(*v1.RoleBinding); ok {
+		return s.create(object)
+	}
+	return nil, shared.HandleBadRequestError(fmt.Sprintf("wrong entity format, attempting roleBinding format, received '%T'", entity))
+}
+
+func (s *service) create(entity *v1.RoleBinding) (*v1.RoleBinding, error) {
+	// TODO: validate user + role exists
+	if len(entity.Spec.Subjects) == 0 {
+		return nil, shared.HandleBadRequestError("spec.subjects can't be empty")
+	}
+
+	// Update the time contains in the entity
+	entity.Metadata.CreateNow()
+	if err := s.dao.Create(entity); err != nil {
+		return nil, err
+	}
+	return entity, nil
+}
+
+func (s *service) Update(entity api.Entity, parameters shared.Parameters) (interface{}, error) {
+	if object, ok := entity.(*v1.RoleBinding); ok {
+		return s.update(object, parameters)
+	}
+	return nil, shared.HandleBadRequestError(fmt.Sprintf("wrong entity format, attempting roleBinding format, received '%T'", entity))
+}
+
+func (s *service) update(entity *v1.RoleBinding, parameters shared.Parameters) (*v1.RoleBinding, error) {
+	if entity.Metadata.Name != parameters.Name {
+		logrus.Debugf("name in roleBinding %q and name from the http request %q don't match", entity.Metadata.Name, parameters.Name)
+		return nil, shared.HandleBadRequestError("metadata.name and the name in the http path request don't match")
+	}
+	if len(entity.Metadata.Project) == 0 {
+		entity.Metadata.Project = parameters.Project
+	} else if entity.Metadata.Project != parameters.Project {
+		logrus.Debugf("project in roleBinding %q and project from the http request %q don't match", entity.Metadata.Project, parameters.Project)
+		return nil, shared.HandleBadRequestError("metadata.project and the project name in the http path request don't match")
+	}
+
+	// TODO: validate user + role exists
+	if len(entity.Spec.Subjects) == 0 {
+		return nil, shared.HandleBadRequestError("spec.subjects can't be empty")
+	}
+
+	// find the previous version of the roleBinding
+	oldEntity, err := s.dao.Get(parameters.Project, parameters.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	// If you do want to change the role for a binding, you need to remove the binding object and create a replacement.
+	// More info at: https://github.com/perses/perses/blob/main/docs/authorization.md#rolebinding-and-globalrolebinding-update-restriction
+	if entity.Spec.Role != oldEntity.Spec.Role {
+		return nil, shared.HandleBadRequestError("spec.role can't be updated")
```

Comment:
I'm wondering if it's a big issue to be able to change the role

Reply:
I found this restriction by looking K8S perm doc and I'm ok with their [reasons](https://github.com/perses/perses/blob/main/docs/authorization.md#rolebinding-and-globalrolebinding-update-restriction)